internal/logger/logger.go

```
@@ -0,0 +1,47 @@
+package logger
+
+import (
+	"context"
+	"log"
+	"time"
+	"userservice/internal/settings"
+
+	"go.mongodb.org/mongo-driver/mongo"
+	"go.mongodb.org/mongo-driver/mongo/options"
+)
+
+type MongoLogger struct {
+	Database *mongo.Database
+}
+
+type Request struct {
+	Date        time.Time
+	Uri         string
+	RequestType string `bson:"request_type"`
+	Err         string
+}
+
+func InitLogger(s *settings.MongoDBSettings) (*MongoLogger, error) {
+	credential := options.Credential{
+		Username: s.User,
+		Password: s.Password,
+	}
+	uri := "mongodb://" + s.Host + ":" + s.Port
+	clientOptions := options.Client().ApplyURI(uri).SetAuth(credential)
+	client, err := mongo.Connect(context.TODO(), clientOptions)
+	if err != nil {
+		log.Fatal(err)
```

why log.Fatal?

---

better to return the err